src/pal/pal_posix.h

```diff
@@ -132,6 +132,26 @@ namespace snmalloc
 
     static constexpr size_t page_size = Aal::smallest_page_size;
 
+    /*
+     * Address bit width is not purely a property of the architecture; it is
+     * potentially mediated by the platform.  By default, though, just pass
+     * through the aal's notion.
+     */
+
+    template<typename T, typename = size_t>
```

Comment:
These look duplicated from the things in the AAL, can they be merged somehow?

Reply:
I played with that for a while and I don't have an easy answer to hand.  The challenge is that SFINAE really does rely on the the S aspect, and C++'s template language doesn't let us quantify over the right kinds to composite things in a context where SFINAE holds.  Less abstractly, C++ templates don't have a first-class notion of a field name, so we can't build an AST node `T::field` in a SFINAE context, so the template name itself must encode the field name being tested.

That is, we'd really like to be able to write something like...
```
template <typename Value, Value def, typename Override, field f, typename _sfinae = int>
class Default
{
  Value v = def;
}

template<typename Value, Value def, typename Override, field f, decltype((void)Override::f, (int)0)>
class Default
{
  Value v = Override::f;
}
```
so that `Override::f` not existing continues to trigger a SF (INAE) in the last template argument, so that the compiler doesn't even see, and complain about, the same bit of syntax at the expression level three lines down.  But we can't, because `field` isn't a quantifiable thing.  And we can't write something like
```
template <typename Value, Value def, Value override, typename _sfinae = int>
class Default
{
  Value v = def;
}

template<typename Value, Value def, Value override, decltype((void)override, (int)0)>
class Default
{
  Value v = override;
}
```
because that requires that we construct our override (i.e., `Override::f`) at the call-site, but if it doesn't exist, that's not a SF, that's just AE.  And since SFINAE applies only at the particular template being considered, we can't try hiding the composition inside another template without basically duplicating the logic.

Less generally, I don't think we want to composite this with the AAL to have a three-layer subsumption architecture in the PAL definition because not all AALs are `AAL_Generic<>`, so we should treat `AAL::address_bits` as opaquely defined over there.